### PR TITLE
Add Package.swift support for tuist edit

### PR DESCRIPTION
**Pull request by fortmarek**
Resolves https://github.com/tuist/tuist/issues/5537

### Short description 📝

Add Package.swift support for tuist edit.

The implementation more-or-less copies the business logic for `Dependencies.swift`. But apart from having a new scheme `Packages` that includes the file `Package.swift`, we also needed to ensure we include the `PackageDescription` framework and we build it with the correct version defined by `swift-tools-version` comment.

That's achieved by:
- Setting `-package-description-version x.y.z` for `OTHER_SWIFT_FLAGS` where `.x.y.z` can be extracted from `swift package tools-version` command
- And `/Applications/Xcode-x.y.z.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/pm/ManifestAPI` `SWIFT_INCLUDE_PATHS`

### How to test the changes locally 🧐

Run `tuist edit` for `fixtures/app_with_spm_dependencies`

### Contributor checklist ✅

- [x] The code has been linted using run `make workspace/lint-fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [x] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase
- [x] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

File: Sources/TuistSupport/SwiftPackageManager/SwiftPackageManagerController.swift [hunk 1]
```diff
@@ -16,6 +16,11 @@ public protocol SwiftPackageManagerControlling {
     ///   - printOutput: When true it prints the Swift Package Manager's output.
     func update(at path: AbsolutePath, printOutput: Bool) throws
 
+    /// Sets tools version of package to the given value.
```
Comment (pepicrft) [inline Sources/TuistSupport/SwiftPackageManager/SwiftPackageManagerController.swift:19]: `Gets the tools version of the package at the given path`

File: Sources/TuistKit/ProjectEditor/ProjectEditorMapper.swift [hunk 2]
```diff
@@ -246,6 +258,30 @@ final class ProjectEditorMapper: ProjectEditorMapping {
...
+                filesGroup: manifestsFilesGroup,
+                targetSettings: Settings(
+                    base: [
+                        "OTHER_SWIFT_FLAGS": .array([
+                            "-package-description-version",
+                            packageVersion.description,
+                        ]),
+                        "SWIFT_INCLUDE_PATHS": .array([
+                            "/Applications/Xcode-\(xcodeVersion).app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/pm/ManifestAPI",
+                        ]),
```
Comment (pepicrft) [inline Sources/TuistKit/ProjectEditor/ProjectEditorMapper.swift:276]: Amazing findings!

**Review by pepicrft — APPROVED** [review]

File: Sources/TuistKit/ProjectEditor/ProjectEditorMapper.swift (hunk 2 above)
Comment (danyf90) [reply Sources/TuistKit/ProjectEditor/ProjectEditorMapper.swift:276]: Isn't this assuming Xcode install path location? For example, I usually just have my latest Xcode installed, and the path is just `/Applications/Xcode.app`.
Wouldn't be better to use the output of `xcode-select --print-path`?

**Review by danyf90 — APPROVED** [review]

File: Sources/TuistKit/ProjectEditor/ProjectEditorMapper.swift (hunk 2 above)
Comment (fortmarek) [reply Sources/TuistKit/ProjectEditor/ProjectEditorMapper.swift:276]: You're right 🤦‍♂️ Yes, I will update this.

**Pull request merged**